### PR TITLE
feat: add math functions SIGN/CBRT/TRUNC

### LINKED ISSUE #790: Add the math functions `SIGN` / `CBRT` / `TRUNC`
### What & why

The math folder `udf/table/math/` currently has only `E`, `Log2`, and `Round` —
noticeably thin. Adding `SIGN` (the sign of a number), `CBRT` (cube root), and
`TRUNC` (truncate to N decimals) is a quick win that meaningfully fills out the
SQL toolbox.

数学函数目录现在只有 `E`、`Log2`、`Round` 三个,明显偏薄。补上 `SIGN`(符号)、
`CBRT`(立方根)、`TRUNC`(截断到指定小数位),花不了多少时间,却能明显补全 SQL 表达力。

### The task

- `sign(x)` → -1 / 0 / 1
- `cbrt(x)` → cube root
- `trunc(x, d)` → truncate (not round) to `d` decimal places

### Where to look

1. Follow `udf/table/math/Round.java`; in the same folder add `Sign.java`,
   `Cbrt.java`, `Trunc.java` (extend `UDF`, add `@Description`).
2. Register them in `BuildInSqlFunctionTable.java`.
3. Add tests under `query/` + `expect/` in `geaflow-dsl-runtime`, covering
   negative numbers, zero, and null.

### Done when

- [ ] Three functions implemented and registered
- [ ] Negative / zero / null edge cases covered
- [ ] `trunc` vs `round` clearly distinguished (truncate ≠ round) — show an example in the PR
- [ ] Tests pass

### PR DESCRIPTION
**Pull request by YinZiBenRun**
## What & Why

  The math folder `udf/table/math/` currently has only E, Log2, and Round — noticeably thin. Adding SIGN (the sign of a
  number), CBRT (cube root), and TRUNC (truncate to N decimals) is a quick win that meaningfully fills out the SQL
  toolbox.

  Resolves #790

  ## Changes

  ### New UDF classes
  - **Sign.java** — `sign(x)` returns -1 for negative, 0 for zero, 1 for positive. Supports Double/Long/Integer input.
  - **Cbrt.java** — `cbrt(x)` returns the cube root of x. Supports Double input.
  - **Trunc.java** — `trunc(x, d)` truncates x to d decimal places (not round). Supports Double/Long/Integer input with
  optional scale parameter.

  ### Registration
  - Added Sign, Cbrt, Trunc to `BuildInSqlFunctionTable.java`
  - Added `cbrt(Double)` and `trunc(Double, Integer)` static methods to `GeaFlowBuiltinFunctions.java` (sign already
  existed)

  ### Tests
  - Added `MathUdfTest.java` — unit tests for Sign, Cbrt, Trunc UDF classes
  - Added `testCbrt()` and `testTrunc()` to `InternalFunctionsTest.java`
  - All tests pass: 20/20

  ## TRUNC vs ROUND — key distinction

  `trunc` uses `RoundingMode.DOWN` while `round` uses `RoundingMode.HALF_UP`:

  ```sql
  -- trunc: truncate (no rounding)
  SELECT trunc(3.1465, 2)  →  3.14
  SELECT trunc(-3.1465, 2) →  -3.14

  -- round: round half up
  SELECT round(3.1465, 2)  →  3.15
  SELECT round(-3.1465, 2) →  -3.15

  Edge cases covered

  ┌──────────┬─────────────────────────────┬────────┬─────────┐
  │ Function │          Negative           │  Zero  │  Null   │
  ├──────────┼─────────────────────────────┼────────┼─────────┤
  │ sign     │ ✅ -1                       │ ✅ 0   │ ✅ null │
  ├──────────┼─────────────────────────────┼────────┼─────────┤
  │ cbrt     │ ✅ -2.0 (cbrt(-8))          │ ✅ 0.0 │ ✅ null │
  ├──────────┼─────────────────────────────┼────────┼─────────┤
  │ trunc    │ ✅ -3.14 (trunc(-3.1465,2)) │ ✅ 0.0 │ ✅ null │
  └──────────┴─────────────────────────────┴────────┴─────────┘

### CODE REVIEW

**Comment by YinZiBenRun** [comment]
#790

**Comment by YinZiBenRun** [comment]
#790

File: geaflow/geaflow-dsl/geaflow-dsl-plan/src/test/java/org/apache/geaflow/dsl/udf/math/MathUdfTest.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import org.apache.geaflow.dsl.udf.table.math.Cbrt;
+import org.apache.geaflow.dsl.udf.table.math.Sign;
+import org.apache.geaflow.dsl.udf.table.math.Trunc;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class MathUdfTest {
+
+    @Test
+    public void testSign() {
```
Comment (Leomrlin) [inline geaflow/geaflow-dsl/geaflow-dsl-plan/src/test/java/org/apache/geaflow/dsl/udf/math/MathUdfTest.java:31]: These tests verify the Java methods directly, but they do not exercise SQL registration, overload resolution, type inference, or runtime invocation.
Could we add an end-to-end SQL test for SIGN/CBRT/TRUNC there?

Comment (YinZiBenRun) [reply geaflow/geaflow-dsl/geaflow-dsl-plan/src/test/java/org/apache/geaflow/dsl/udf/math/MathUdfTest.java:31]: 已添加端到端 SQL 测试，通过完整查询管线验证函数的注册、重载解析、类型推断和运行时调用。测试在 geaflow-dsl-runtime 模块：

  - MathSignCbrtTruncTest.java - 通过 QueryTester 驱动测试
  - query/math_sign_cbrt_trunc_001.sql - 覆盖 sign(Double/Long/Integer)、cbrt(Double)、trunc(Double 带/不带 scale、Long、Integer)
  - data/ + expect/ - 输入数据与期望输出

  测试揭示了 Calcite 真实的重载解析行为：sign(bigint)/sign(int) 会被提升为 double，匹配 sign(Double) 重载（返回 1.0/-1.0）；而
  trunc(bigint)/trunc(int) 则精确匹配 Long/Integer 重载。期望输出里记录了这个差异。

  已在本地验证通过（Tests run: 1, Failures: 0, Errors: 0）。

**Review by Leomrlin — APPROVED** [review]
LGTM
